docs/docs/guides/next.mdx

```diff
@@ -82,85 +82,154 @@ export default function Home() {
 }
 ```
 
-### Define login callbacks
+## Define event callbacks
 
-The `<hanko-auth>` element dispatches a custom `hankoAuthSuccess` event on successful login. React to this
-event in order to, for example, redirect your users to protected pages in your application,
-e.g. a [user profile page](#hanko-profile).
+Use the Hanko client provided by `@teamhanko/hanko-elements` to subscribe to
+[events](https://github.com/teamhanko/hanko/blob/main/frontend/elements/README.md#events).
 
-To do so, apply an event listener with an appropriate redirect callback:
-
-```jsx {2,10-19} title="HankoAuth.jsx" showLineNumbers
-import React, { useEffect, useCallback } from "react";
+```jsx title="HankoAuth.jsx" showLineNumbers
+import React, { useEffect, useCallback, useState } from "react";
+// highlight-start
 import { useRouter } from "next/router";
-import { register } from "@teamhanko/hanko-elements";
+import { register, Hanko } from "@teamhanko/hanko-elements";
+// highlight-end
 
+// highlight-next-line
 const hankoApi = "<YOUR_API_URL>";
 
 export default function HankoAuth() {
+  // highlight-next-line
   const router = useRouter();
 
+  // highlight-start
+  const [hanko, setHanko] = useState<Hanko>();
+
+  useEffect(() => {
+    import("@teamhanko/hanko-elements").then(({ Hanko }) => setHanko(new Hanko(hankoApi)));
```

Comment:
When I change the code to:
```
import { Hanko } from "@teamhanko/hanko-elements";

// ...

const Todo: NextPage = () => {
  // ...
  const [hankoClient, setHankoClient] = useState<Hanko>();
  // ...

  useEffect(() => {
    setHankoClient(new Hanko(hankoAPI)));
  }, []);

  // ...
}
```

It will result in an error "ReferenceError: CustomEvent is not defined". I think it's caused by a class declaration in the SDK that extends the native `CustomEvent`.

```
export class CustomEventWithDetail<T> extends CustomEvent<T> {
 // ...
}
```

But I couldn't find a way to overcome this issue, plus I am not 100% sure what exactly happens within the next.js renderer. Also there are possibly more issues with SSR. Let me know if you have any idea, would love to see it working without the dynamic import.

Reply:
Ok, I can't think of anything either at the moment.